### PR TITLE
Standardize exports in constants.js

### PR DESCRIPTION
**Pull request by islemaster**
We are inconsistent about our module format, and it bugs me, but it's too large of a project to do all in one go.

I figure a good starting point is to update some widely-used modules and also every place that imports them.  This is likely to establish the beginning of a good pattern in a lot of places (and minimize how often we have to parse old module syntax) and lead the team toward cleaning this up over time.

Starting with `[cdo]/apps/constants.js` here.

### CODE REVIEW

**Comment by islemaster** [comment]
CSF team: It almost looks like you've specifically avoided this in some of your code. Am I stepping on some decision you've made here?

**Comment by Hamms** [comment]
@joshlory ^

**Comment by joshlory** [comment]
My issue with `import` is that there's [currently no JS runtime in existence that implements it](https://medium.com/the-node-js-collection/an-update-on-es6-modules-in-node-js-42c958b890c).  JavaScript is intended to be an interpreted language, but now we're _requiring_ a pre-compilation step.

That's why in the [craft](https://github.com/code-dot-org/craft) repo we use `tape` as a test runner.  I'm a big fan of the [tap philosophy](https://www.node-tap.org/):

> **Test files should be “normal” programs that can be run directly.**
>
> That means that it can’t require a special runner that puts magic functions into a global space. `node test.js` is a perfectly ok way to run a test, and it ought to function exactly the same as when it’s run by the fancy runner with reporting and such. JavaScript tests should be JavaScript programs; not english-language poems with weird punctuation.
>
> [...]
>
> **Tests should not require more building than your code.**
>
> Babel and Webpack are lovely and fine. But if your code doesn’t require compilation, then I think your tests shouldn’t either.

**All that said:**  For this repo I don't think it's something we can address.  We're heavily invested in Mocha, which [wants to be a full-featured test runner](https://medium.com/javascript-scene/why-i-use-tape-instead-of-mocha-so-should-you-6aa105d8eaf4).

**Comment by joshlory** [comment]
TL;DR: no concerns whatsoever with changing `require` -> `import` within the `code-dot-org` repo — we should totally standardize on one format, and for us `import` probably makes the most sense!  I just offer this background for why I prefer `require` in some other repo configurations 😄.

This isn't something I feel super strongly about.  We break this rule in the [maze](https://github.com/code-dot-org/maze) because, well... Jest is awesome.  But maybe that's how folks used to feel about Mocha too.

**Comment by islemaster** [comment]
> My issue with import is that there's currently no JS runtime in existence that implements it.

That's changed a lot in the past year.

[Browser support has gotten a lot better.](https://caniuse.com/#feat=es6-module) Chrome [shipped support for native ES modules in v61 in September](https://medium.com/dev-channel/es6-modules-in-chrome-canary-m60-ba588dfb8ab7).  Apparently [Edge's Chakra engine has supported them a lot longer](https://blogs.windows.com/msedgedev/2016/05/17/es6-modules-and-beyond/). Safari introduced full support in 11, and Firefox will have it in the next version, 60.

[NodeJS is on its way to full support](https://medium.com/@giltayar/native-es-modules-in-nodejs-status-and-future-directions-part-i-ee5ea3001f71) having [added support behind the `--experimental-modules` flag in 8.5.0 in September](https://nodejs.org/en/blog/release/v8.5.0/).  That flag [is still present in 10.0.0](https://nodejs.org/api/esm.html) but is supposed to be on by default when Node 10 becomes the next LTS version in October.

I'm not opposed to TAP protocol, but as you say, we're a long way from doing that in this repo - and by the time we got around to it, TAP with ES module syntax will probably be an option! :tada:

In the meantime I'm in favor of standardizing on ES module syntax.  It's most likely to be future-proof since it's a native standard, it's a nice syntax, and since it must be used in a static context (unlike `require()`) it will eventually make static analysis and optimizations easier for the tools we use. (See [Benefits of ECMAScript 6 modules](http://2ality.com/2014/09/es6-modules-final.html#benefits-of-ecmascript-6-modules) and [the Chakra blog post](https://blogs.windows.com/msedgedev/2016/05/17/es6-modules-and-beyond/) for more about this.)

That does mean we should probably be moving _away_ from some of the special module resolution rules we've configured in webpack (`@cdo/apps`, etc.) but again, that's a long-term effort and it doesn't hurt to switch to `import/export` syntax now since we support it.

**Comment by islemaster** [comment]
> but now we're requiring a pre-compilation step.

The other limiting factor here has less to do with our module syntax and more to do with our npm dependencies, which AFAIK would still need to be bundled somehow and provided in a native module context.

**Comment by islemaster** [comment]
Just making sure: Using ES module syntax in `code-dot-org` does not require you to change any of your decisions for the `craft` repo, correct?  I'm guessing no, but wanted to double check.

**Comment by joshlory** [comment]
> Just making sure: Using ES module syntax in `code-dot-org` does not require you to change any of your decisions for the `craft` repo, correct?

Correct!

> That's changed a lot in the past year.

TIL!  Very exciting — looking forward to being able to use `import` everywhere 🎉.

**Comment by islemaster** [comment]
ESLint's [`object-curly-spacing`](https://eslint.org/docs/rules/object-curly-spacing) rule handles this situation (and is autofixable!) but we've [intentionally disabled it](https://github.com/code-dot-org/code-dot-org/blob/59345c776721a49a908b29aea0c0fb2cc7851e1b/.eslintrc.js#L44). ([Background](https://github.com/code-dot-org/code-dot-org/pull/7678))

Since this is a style rule that don't have a potential impact on behavior we've left it unset because enabling it would actually _add_ cognitive load to devs.  However, this is on the list of things I'd _love_ to standardize on as soon as we have seamless autofix working, rather than something you have to run manually before your commit.
